### DISCUSSION TITLE
Translating component libraries

### DISCUSSION THREAD

**Discussion by benmvp** [i18next/next-i18next#2152] [Q&A]
I searched through the Issues & Discussions and found nothing relevant...

My app uses a component library (separate package) that I also want to be translated as well. Since `appWithTranslation` wraps the whole app with the I18nContext I should have access to the locale, call `useTanslation`, etc.

But where do I put the JSON files for the component? For the app they're in the `/public` folder, but any component library would have the files deep in `node_modules`. And how would I reference them? I imagine the same thing would happen in a monorepo as well.

I appreciate any direction anyone can give! 🙏🏾

**Comment by adrai** [comment]
Maybe have the translations loaded in-memory like here: https://github.com/i18next/react-i18next/tree/master/example/react-component-lib
?
